core/src/main/java/bisq/core/dao/burningman/BurningManService.java

```diff
@@ -0,0 +1,638 @@
+            if (available > 100000) {
+                receivers.add(new Tuple2<>(available, getBurningManAddressFromParam(burningManSelectionHeight)));
+            }
+        }
+        return receivers;
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Utils
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    private int getCycleIndex(int height) {
+        return daoStateService.getCycle(height).map(cycleService::getCycleIndex).orElse(0);
+    }
+
+    private static long getSpendableAmount(int numOutputs, long inputAmount, long txFeePerVbyte) {
+        // Output size: 32 bytes
+        // Tx size without outputs: 51 bytes
+        int txSize = 51 + numOutputs * 32;
+        long minerFee = txFeePerVbyte * txSize;
+        return inputAmount - minerFee;
+    }
+
+    private long getDecayedCompensationAmount(long amount, int issuanceHeight, int chainHeight) {
+        return getDecayedAmount(amount, issuanceHeight, chainHeight, chainHeight - MAX_COMP_REQUEST_AGE, 0);
+    }
+
+    private long getDecayedGenesisOutputAmount(long amount) {
+        return Math.round(amount * GENESIS_OUTPUT_AMOUNT_FACTOR);
+    }
+
+    private long getDecayedBurnedAmount(long amount, int issuanceHeight, int chainHeight) {
+        return getDecayedAmount(amount,
+                issuanceHeight,
+                chainHeight,
+                chainHeight - MAX_BURN_AMOUNT_AGE,
+                0);
+    }
+
+    // Linear decay between currentBlockHeight (100% of amount) and issuanceHeight (firstBlockOffset % of amount)
+    // Values below firstBlockHeight will use the firstBlockOffset as factor for the amount.
+    // E.g. if firstBlockOffset is 0.1 the decay goes to 10% and earlier values stay at 10%.
+    @VisibleForTesting
+    static long getDecayedAmount(long amount,
+                                 int issuanceHeight,
+                                 int currentBlockHeight,
+                                 int firstBlockHeight,
+                                 double firstBlockOffset) {
+        if (issuanceHeight > currentBlockHeight)
+            throw new IllegalArgumentException("issuanceHeight must not be larger than currentBlockHeight. issuanceHeight=" + issuanceHeight + "; currentBlockHeight=" + currentBlockHeight);
+        if (currentBlockHeight < 0)
+            throw new IllegalArgumentException("currentBlockHeight must not be negative. currentBlockHeight=" + currentBlockHeight);
+        if (amount < 0)
+            throw new IllegalArgumentException("amount must not be negative. amount" + amount);
+        if (issuanceHeight < 0)
+            throw new IllegalArgumentException("issuanceHeight must not be negative. issuanceHeight=" + issuanceHeight);
+
+        double factor = Math.max(0, (issuanceHeight - firstBlockHeight) / (double) (currentBlockHeight - firstBlockHeight));
```

Comment:
We are doing that already because the firstBlockHeight is the currentCycleHeight - numCycles*cycleDuration.
I rename it and add a comment to make it more clear.
See cyclesInDaoStateService.getChainHeightOfPastCycle